test/TripleSerializerTest.cpp

```diff
@@ -142,4 +142,25 @@ TEST(TripleSerializer, onlySingleWordSetSupportedForLocalVocab) {
   EXPECT_THROW(ad_utility::detail::serializeLocalVocab(serializer, localVocab),
                ad_utility::Exception);
 }
+
+// _____________________________________________________________________________
+TEST(TripleSerializer, rethrowsOnInvalidFileAccess) {
+  using namespace ::testing;
+  auto tmpFile = std::filesystem::temp_directory_path() / "fileNoPermissions";
+  // Create empty file
+  std::ofstream{tmpFile}.close();
+  absl::Cleanup cleanup{[&tmpFile]() { std::filesystem::remove(tmpFile); }};
+  // Remove all permissions to make read fail
+  std::filesystem::permissions(tmpFile, std::filesystem::perms::none);
```

Comment:
Won't this make the `remove` call also fail? and what happens if the remove fails, and I run the tests twice while the tmp file is still present, and I don't have the permissions to set the permissions?